pywhy_graphs/algorithms/generic.py

```diff
@@ -703,3 +703,56 @@ def valid_mag(G: ADMG, L: set = None, S: set = None):
                 return False
 
     return True
+
+
+def dag_to_mag(G, L: Set = None, S: Set = None):
+    """Converts a DAG to a valid MAG.
+    The algorithm is defined in :footcite:`Zhang2008`.
```

Comment:
The page numbering is very weird. It's in the thousands and it's not defined as a theorem. I will just put the page number. (although it will look weird)